cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -101,14 +104,33 @@ pub struct YaccGrammar {
 // Internally, we assume that a grammar's start rule has a single production. Since we manually
 // create the start rule ourselves (without relying on user input), this is a safe assumption.
 
-impl YaccGrammar {
-    /// Translate a `GrammarAST` into a `YaccGrammar`. This function is akin to the part a traditional
-    /// compiler that takes in an AST and converts it into a binary.
+impl YaccGrammar<u32> {
+    pub fn new(yacc_kind: YaccKind, s: &str) -> Result<Self, YaccGrammarError> {
+        YaccGrammar::new_with_storaget(yacc_kind, s)
+    }
+}
+
+impl<StorageT: PrimInt + Unsigned> YaccGrammar<StorageT> {
+    /// Takes as input a Yacc grammar of [`YaccKind`](enum.YaccKind.html) as a `String` `s` and returns a
+    /// [`YaccGrammar`](grammar/struct.YaccGrammar.html) (or
+    /// ([`YaccGrammarError`](grammar/enum.YaccGrammarError.html) on error).
     ///
-    /// As we're compiling the `GrammarAST` into a `Grammar` we add a new start rule (which we'll
-    /// refer to as `^`, though the actual name is a fresh name that is guaranteed to be unique)
-    /// that references the user defined start rule.
-    pub fn new(yacc_kind: YaccKind, ast: &ast::GrammarAST) -> YaccGrammar {
+    /// As we're compiling the `YaccGrammar`, we add a new start rule (which we'll refer to as `^`,
+    /// though the actual name is a fresh name that is guaranteed to be unique) that references the
+    /// user defined start rule.
+    pub fn new_with_storaget(yacc_kind: YaccKind, s: &str)
+                          -> Result<Self, YaccGrammarError>
+    {
+        let ast = match yacc_kind {
+            YaccKind::Original | YaccKind::Eco => {
```

Comment:
If there is only one "arm" in this `match`, is there any reason not to say `match yacc_kind {  _ => ... };`?

Reply:
Yes, because we *might* in future add `YaccKind`s which can't use this code path, and then we'll get a static type error reminding us to fix things.

Reply:
OK